Retirement of SSH over Local Companion in VS Code Desktop changelog (#4002)
gitpod-io/website: src/content/changelog/2023-10-05-local-app-retired-in-vscode-desktop.md

---
title: Local Companion (as an SSH connection method) for VS Code Desktop is now retired
alt: Image banner for Retirement of SSH over Local Companion in VS Code Desktop

> date: 2023-10-06

Author: also in the name of a file

excerpt: We're streamlining your VS Code Desktop experience by retiring SSH over Local Companion.

> image: retirement-ssh-local-companion-in-vscode-dekstop.webp

Reviewer: This image seems unrelated.

ogImage: vscode.webp
customSlug: local-companion-ssh-connection-retired
---

> Over the past few months we have continued to introduce more stable and reliable methods of connecting to workspaces via SSH. The previously supported method of establishing an SSH connection using the Local Companion app is now retired. No action is required by users, VS Code will automatically switch to the new connection method. For troubleshooting, consult our [guide](/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop).

Author: `/docs/references/ides-and-editors/vscode#troubleshooting` as a guide

---

> title: Retirement of SSH over Local Companion in VS Code Desktop

Reviewer: Suggestion: Might be better to use "deprecated" rather than "retirement"?

```suggestion
title: Local Companion (SSH) for VS Code Desktop is now deprecated
```

---

> title: Retirement of SSH over Local Companion in VS Code Desktop

Author: This changelog to highlight that we don't support Local Companion mode anymore.

> title: Retirement of SSH over Local Companion in VS Code Desktop

Author: i.e we still do HTTP tunnel but differently

alt: Image banner for Retirement of SSH over Local Companion in VS Code Desktop
date: 2023-10-05
excerpt: We're streamlining your VS Code Desktop experience by retiring SSH over Local Companion.
image: retirement-ssh-local-companion-in-vscode-dekstop.jpg
ogImage: retirement-ssh-local-companion-in-vscode-dekstop.jpg
customSlug: retirement-of-ssh-over-local-companion-in-vs-code-desktop
---

> -   **Streamlined Experience**: We're focusing on offering you a single, reliable mode for connecting VS Code Desktop with Gitpod. Details can be found in our [updated documentation](/docs/references/ides-and-editors/vscode#connecting-to-vs-code-desktop). Ensure you're using the latest versions of VS Code, Remote SSH, and the Gitpod extension for the most seamless experience.

Reviewer 2: **nitpick**: Does this need to be also in a list? I'd expect to read something, the main message, as a plain paragraph even if it's just 2-3 sentences.

-   **Legacy Mode**: SSH Gateway will remain available as a fallback for now. SSH over Local Companion is being retired specifically for VS Code Desktop.

-   **Local Companion**: The Local Companion feature will continue to be supported for use with VS Code Browser.

## No Action Required

> If you've been using SSH over Local Companion, your setup will automatically switch to the optimized mode. For troubleshooting, consult our [guide](/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop).

Reviewer 2: **praise**: Nice help link!

> We appreciate your ongoing support and look forward to enhancing your Gitpod experience! 🎉

Reviewer 2: **thought**: Not sure about how to feel about this line. 🤷
